src/advanced_unsafety/invalid_values.md

```diff
@@ -0,0 +1,123 @@
+# Invalid values
+
+> _“If you tell the truth, you don't have to remember anything.”__
+>  — _Mark Twain_
+
+Values of a particular type in Rust may never have an "invalid" bit pattern for that type. This is true even if that value is never read from afterwards.
```

Comment:
We should mirror some of the wording from ["Behavior considered undefined"](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) here. Specifically how "producing an invalid value" travels along references since references must not point to invalid values.